File: samples/client/wordnik-api/python/tests/WordApiTest.py
#!/usr/bin/env python

import sys
import unittest
import urllib2
import json

from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from wordnik import *


class WordApiTest(BaseApiTest):

    def testWordApis(self):
        response = urllib2.urlopen('http://api.wordnik.com/v4/word.json')
        doc = json.loads(response.read())
        assert len(doc['apis']) == 12, 'there should be 10 word apis'

    def testGetWord(self):
        res = self.wordApi.getWord('cat')
        assert res, 'null getWord result'
        assert res.word == 'cat', 'word should be "cat"'

    def testGetWordWithSuggestions(self):
        res = self.wordApi.getWord('cAt', includeSuggestions=True)
        assert res, 'null getWord result'
        assert res.word == 'cAt', 'word should be "cAt"'

    def testGetWordWithCanonicalForm(self):
        res = self.wordApi.getWord('cAt', useCanonical=True)
        assert res, 'null getWord result'
        assert res.word == 'cat', 'word should be "cAt"'

    def testGetDefinitions(self):
        res = self.wordApi.getDefinitions('cat', limit=10)
        assert res, 'null getDefinitions result'
        assert len(res) == 10, 'should have 10 definitions'

    def testGetDefinitionsWithSpacesInWord(self):
        res = self.wordApi.getDefinitions('bon vivant')
        assert res, 'null getDefinitions result'
        assert len(res) == 1, 'should have 1 definition'

    def testGetExamples(self):
        res = self.wordApi.getExamples('cat', limit=5)
        assert res, 'null getExamples result'
        assert len(res.examples) == 5, 'should have 5 definitions'

    def testGetTopExample(self):
        res = self.wordApi.getTopExample('cat')
        assert res, 'null getTopExample result'
        assert res.word == 'cat', 'word should be "cat"'

    def testGetHyphenation(self):
        res = self.wordApi.getHyphenation('catalog', limit=1)
        assert res, 'null getHyphenation result'
        assert len(res) == 1, 'hypenation length should be 1'

    def testGetWordFrequency(self):
        res = self.wordApi.getWordFrequency('cat')
        assert res, 'null getWordFrequency result'
        assert res.totalCount != 0, 'total count should not be 0'

    def testGetPhrases(self):
        res = self.wordApi.getPhrases('money')
        assert res, 'null getPhrases result'
        assert len(res) != 0, 'getPhrases length should not be 0'

    def testGetRelatedWords(self):
        res = self.wordApi.getRelatedWords('cat')
        assert res, 'null getRelatedWords result'
        for related in res:
            assert len(related.words) <= 10, 'should have <= 10 related words'

    def testGetAudio(self):
        res = self.wordApi.getAudio('cat', useCanonical=True, limit=2)
        assert res, 'null getAudio result'
        assert len(res) == 2, 'getAudio size should be 2'

    def testGetScrabbleScore(self):
        res = self.wordApi.getScrabbleScore('quixotry')
        assert res.value == 27, 'quixotry should have a Scrabble score of 27'

    def testGetEtymologies(self):
        res = self.wordApi.getEtymologies('butter')
        assert 'of Scythian origin' in res[0], 'etymology of "butter" should contain the phrase "of Scythian origin"'


if __name__ == "__main__":
    unittest.main()

File: samples/client/wordnik-api/php/tests/WordApiTest.php
<?php

require_once 'BaseApiTest.php';

date_default_timezone_set('America/Los_Angeles');

class WordApiTest extends BaseApiTest {

  public function testWordApis() {
    $ch = curl_init("http://api.wordnik.com/v4/word.json");
    if (! $ch) {
    	die("No php curl handle");
    }
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);

    $data = curl_exec($ch);
    $doc = json_decode($data);

    $this->assertEquals(12, count($doc->apis));
  }

  public function testGetWord() {
    $res = $this->wordApi->getWord('cat');
    $this->assertEquals('cat', $res->word);
  }

  public function testGetWordWithSuggestions() {
    $res = $this->wordApi->getWord('cAt', $includeSuggestions=true);
    $this->assertEquals('cAt', $res->word);
  }

  public function testGetWordWithCanonicalForm() {
    $res = $this->wordApi->getWord('cAt', $useCanonical='true');
    $this->assertEquals('cat', $res->word);
  }

  public function testGetDefinitions() {
    $res = $this->wordApi->getDefinitions('cat');
    $this->assertEquals(15, count($res));
  }

  public function testGetDefinitionsWithSpacesInWord() {
    $res = $this->wordApi->getDefinitions('bon vivant');
    $this->assertEquals(1, count($res));
  }

  public function testGetExamples() {
    $res = $this->wordApi->getExamples('cat', $limit=5);
    $this->assertEquals(5, count($res->examples));
  }

  public function testGetTopExample() {
    $res = $this->wordApi->getTopExample('cat', $limit=5);
    $this->assertEquals('cat', $res->word);
  }

  public function testGetHyphenation() {
    $res = $this->wordApi->getHyphenation('catalog', $useCanonical=null, $sourceDictionary=null, $limit=1);
    $this->assertEquals(1, count($res));
  }

  public function testGetWordFrequency() {
    $res = $this->wordApi->getWordFrequency('catalog');
    $this->assertFalse($res->totalCount == 0);
  }

  public function testGetPhrases() {
    $res = $this->wordApi->getPhrases('money');
    $this->assertFalse(count($res) == 0);
  }

  public function testGetRelatedWords() {
    $res = $this->wordApi->getRelatedWords('cat');
    foreach ($res as $related) {
      $this->assertLessThan(11, count($related->words));
    }
  }

  public function testGetAudio() {
    $res = $this->wordApi->getAudio('cat', $useCanonical=True, $limit=2);
    $this->assertEquals(2, count($res));
  }

  public function testGetScrabbleScore() {
    $res = $this->wordApi->getScrabbleScore('quixotry');
    $this->assertEquals(27, $res->value);
  }

  public function testGetEtymologies() {
    $res = $this->wordApi->getEtymologies('butter');
    $this->assertFalse(strpos($res[0], 'of Scythian origin') === false);
  }

}
?>
